src/cpu/pred/BranchPredictor.py

```diff
@@ -1088,3 +1088,12 @@ class MultiperspectivePerceptronTAGE8KB(MultiperspectivePerceptronTAGE):
     tage = MPP_TAGE_8KB()
     loop_predictor = MPP_LoopPredictor_8KB()
     statistical_corrector = MPP_StatisticalCorrector_8KB()
+
+
+class GshareBP(BranchPredictor):
+    type = "GshareBP"
+    cxx_class = "gem5::branch_prediction::GshareBP"
+    cxx_header = "cpu/pred/gshare.hh"
+
+    globalPredictorSize = Param.Unsigned(512, "Size of global predictor")
+    globalCtrBits = Param.Unsigned(2, "Bits per counter")
```

Comment:
Two suggestions: (1) use snake case since this is python, and (2) use full words to reduce the possibilities of confusion later.

```suggestion
    global_predictor_size = Param.Unsigned(512, "Size of global predictor")
    global_counter_bits = Param.Unsigned(2, "Bits per counter")
```

Reply:
Done